components/core/cmake/Modules/FindLibArchive.cmake

```diff
@@ -20,10 +20,19 @@ include(cmake/Modules/FindLibraryDependencies.cmake)
 find_package(PkgConfig)
 pkg_check_modules(libarchive_PKGCONF QUIET "lib${libarchive_LIBNAME}")
 
+# On macOS, libarchive installed through brew is not linked into prefix by default. 
+# So it cannot be found by pkg-config and we need to manually find it.
+# For more details, see https://github.com/Homebrew/homebrew-core/issues/117642
+if(APPLE)
+    execute_process(COMMAND brew --prefix libarchive OUTPUT_VARIABLE PREFIX_LIBARCHIVE)
```

Comment:
Ah, good catch. I added this to the [GitHub action](https://github.com/y-scope/clp/blob/main/.github/workflows/clp-core-build.yaml#L129) but forgot to document it for users. The way I do it in the workflow is by setting [CMAKE_PREFIX_PATH](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html#variable:CMAKE_PREFIX_PATH), so I wonder if instead of adding the prefix to each find command below we could add the prefix to `CMAKE_PREFIX_PATH` in `CMakeLists.txt`. Something like:

```
# Find and setup libarchive
if(APPLE)
    execute_process(COMMAND brew --prefix libarchive OUTPUT_VARIABLE libarchive_MACOS_PREFIX)
    string(STRIP "${libarchive_MACOS_PREFIX}" libarchive_MACOS_PREFIX)
    set(ENV{CMAKE_PREFIX_PATH} "${libarchive_MACOS_PREFIX};$ENV{CMAKE_PREFIX_PATH}")
endif()
```

Reply:
Good idea! `CMAKE_PREFIX_PATH` looks more concise. Updated.